src/pubsub_test/pubsub_sync_client.clj

```diff
@@ -0,0 +1,179 @@
+(ns pubsub-test.pubsub-sync-client
+  (:require [taoensso.timbre :as log])
+  (:import [com.google.cloud.pubsub.v1 SubscriptionAdminSettings]
+           [com.google.cloud.pubsub.v1.stub GrpcSubscriberStub]
+           [com.google.pubsub.v1
+            SubscriptionName
+            PullRequest
+            ModifyAckDeadlineRequest
+            AcknowledgeRequest]
+           [org.threeten.bp Duration]
+           [java.util.concurrent
+            Executors
+            ScheduledExecutorService
+            TimeUnit]
+           [com.google.common.util.concurrent
+            MoreExecutors
+            Futures
+            FutureCallback]
+           java.util.ArrayList
+           java.io.PrintStream
+           org.apache.log4j.MDC
+           org.slf4j.bridge.SLF4JBridgeHandler))
+
+(defn- capture-console-messages!
+  []
+  (System/setOut (proxy [PrintStream] [System/out]
+                   (print [s]
+                     (log/info (str "Console message: " s))))))
+
+(defn- init-logging!
+  []
+  (capture-console-messages!)
+  (let [timbre-config (assoc log/example-config :level :info)]
+    (SLF4JBridgeHandler/removeHandlersForRootLogger)
+    (SLF4JBridgeHandler/install)
+    (log/set-config! timbre-config)))
+
+(defn client-settings
+  [max-rpc-seconds]
+  (let [client-settings-builder (SubscriptionAdminSettings/newBuilder)
+        pull-retry-settings (-> client-settings-builder
+                                (.pullSettings)
+                                (.getRetrySettings)
+                                (.toBuilder)
+                                (.setInitialRpcTimeout (Duration/ofSeconds 1))
+                                (.setMaxAttempts 1)
+                                (.build))
+        ack-retry-settings (-> client-settings-builder
+                               (.acknowledgeSettings)
+                               (.getRetrySettings)
+                               (.toBuilder)
+                               (.setInitialRpcTimeout (Duration/ofSeconds 1))
+                               (.setTotalTimeout (Duration/ofSeconds max-rpc-seconds))
+                               (.build))]
+    (-> client-settings-builder
+        (.pullSettings)
+        (.setRetrySettings pull-retry-settings))
+    (-> client-settings-builder
+        (.acknowledgeSettings)
+        (.setRetrySettings ack-retry-settings))
+    (-> client-settings-builder
+        (.modifyAckDeadlineSettings)
+        (.setRetrySettings ack-retry-settings))
+    (.build client-settings-builder)))
+
+(defn pull-request
+  [subname num-of-messages]
+  (.build (doto (PullRequest/newBuilder)
+            (.setReturnImmediately Boolean/TRUE)
+            (.setSubscription subname)
+            (.setMaxMessages num-of-messages))))
+
+(defn process-messages
```

Comment:
This might seem trivial, but can we move this fn below `mod-ack-request` `send-mod-acks`, `ack-request` and `send-acks`? I know it's not required by the compiler, but this fn depends on those ones (because it sends work to them by way of a queue), so it messed with my understanding of how a clojure file is organised!